src/main/java/bitflow4j/steps/database/WriteHistogramToElasticsearch.java

```diff
@@ -0,0 +1,89 @@
+package bitflow4j.steps.database;
+
+import bitflow4j.Sample;
+import bitflow4j.steps.BatchHandler;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.support.replication.ReplicationResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WriteHistogramToElasticsearch implements BatchHandler {
+
+    protected static final Logger logger = Logger.getLogger(WriteHistogramToElasticsearch.class.getName());
+
+    private final String indexName;
+    private final String identifierKey;
+    private final String identifierTemplate;
+    private final RestHighLevelClient client;
+    public WriteHistogramToElasticsearch(String host, int port1, int port2, String indexName, String identifierKey, String identifierTemplate) {
+        this.indexName = indexName;
+        this.identifierKey = identifierKey;
+        this.identifierTemplate = identifierTemplate;
+        this.client = new RestHighLevelClient(
+                RestClient.builder(
+                        new HttpHost(host, port1, "http"),
```

Comment:
Why are there exactly two ports for this? Should the number of ports maybe be variable? In that case, you can add a constructor that takes an int[] parameter, and a second constructor (to be used in Bitflow Script) that takes a string, splits it on "," comma characters, and then parses the split parts to integers. This should be extracted into a utility method like parseIntArray(String str)

Reply:
I would personally prefer to define a list of host:port pairs, if that is possible. This could also be done by splitting a comma-separated list of strings. In this case, there should be a HttpHost() constructor that takes a single string.